envault/cli.py

```diff
@@ -33,13 +35,56 @@ def get_secrets(server, secret, token):
 def cli():
     pass
 
+@cli.command("init")
+@click.option("-profile", help="profile for the config")
+def init(profile):
+    if profile is None:
+        profile = 'default'
+    """ Initialize envault config with vault server, token and secrets path """
+    click.echo("Enter the server, token and path to vault secrets")
+    vault_server = click.prompt('Vault Server', type=str)
+    vault_token = click.prompt('Vault Token', type=str)
+    vault_secret_path = click.prompt('Path to vault secret', type=str)
+
+    click.echo("server %s token %s path %s" %(vault_server, vault_token, vault_secret_path))
+
+    new_profile = {
+        profile: {
+            'vault_server': vault_server,
+            'vault_token': vault_token,
+            'vault_secret_path': vault_secret_path
+        }
+    }
+
+    if os.path.exists('envault.yml'):
+        yaml_file = open('envault.yml', 'r')
+        all_profile_configs = safe_load(yaml_file)
+        if all_profile_configs is not None and profile in all_profile_configs:
+            yaml_file = open('envault.yml', 'w')
+            updated_profile = {
+                **all_profile_configs,
+                **new_profile
+            }
+
+            dump(updated_profile, yaml_file, default_flow_style = False)            
+        else:
+            yaml_file = open('envault.yml', 'a')
+            dump(new_profile, yaml_file, default_flow_style = False)
+    else:
+        yaml_file = open('envault.yml', 'w+')
+        dump(new_profile, yaml_file, default_flow_style = False)
+
 
 @cli.command("list")
 @click.option("-server", help="Server URI")
 @click.option("-secret", help="Path to the secrets")
 @click.option("-token", help="Vault token")
 def list(server, secret, token):
     """ List secrets from a given path """
+    config = vault.get_profile_configs()
+    server = config.get('vault_server')
+    secret = config.get('vault_secret_path')
+    token = config.get('vault_token')
```

Comment:
Instead of taking configuration each time in every function, create a single function to handle all these.

It's already handled in `get_secrets`, you can add more logic there